app/components/calculate-time.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { Ruleset } from "../interfaces/Ruleset";
+import { EnterHours } from "../interfaces/EnterHours";
+import { calculateHours, getTimeValue } from '../utils/utils';
+
+interface CalculateTimeProps {
+  rulesets: Ruleset[];
+  hoursWorked: EnterHours[];
+}
+
+export function calculateTotalTOIL(rulesets: Ruleset[], hoursWorked: EnterHours[]): number {
+  let totalTOIL = 0;
+  hoursWorked.forEach((hours) => {
+    const hoursDayOfWeek = hours.date.toLocaleString('en-US', { weekday: 'long' });
+    const matchingRuleset = rulesets.find((ruleset) =>
+      // Compare ruleset dayOfWeek and startTime with hours date and startTime
+      ruleset.dayOfWeek === hoursDayOfWeek &&
+      getTimeValue(ruleset.startTime) <= getTimeValue(hours.startTime) &&
+      getTimeValue(ruleset.endTime) >= getTimeValue(hours.endTime)
+    );
```

Comment:
**issue (non-blocking):** This only works if the `hours` sit within a ruleset but do not extend it.

For example, a ruleset of 1.5x between 9:00AM-5:00PM and hours worked of 10:00AM-6:00PM, shows 0 hours worked.

Reply:
**suggestion:** Check for an overlap and apply the multiplier only to that overlap.

Here's some sample code which should give a rough idea of how to approach:

```typescript
export function calculateTotalTOIL(rulesets: Ruleset[], hoursWorked: EnterHours[]): number {
  let totalTOIL = 0;
  hoursWorked.forEach((hours) => {
    const hoursDayOfWeek = hours.date.toLocaleString('en-US', { weekday: 'long' });
    const matchingRuleset = rulesets.find((ruleset) =>
      ruleset.dayOfWeek === hoursDayOfWeek &&
      getTimeValue(ruleset.endTime) > getTimeValue(hours.startTime) &&
      getTimeValue(ruleset.startTime) < getTimeValue(hours.endTime)
    );

    if (matchingRuleset) {
      const overlapStart = Math.max(getTimeValue(matchingRuleset.startTime), getTimeValue(hours.startTime));
      const overlapEnd = Math.min(getTimeValue(matchingRuleset.endTime), getTimeValue(hours.endTime));

      const overlappingHours = calculateHoursFromValues(overlapStart, overlapEnd); // Assuming you'd need a new function for this
      totalTOIL += overlappingHours * matchingRuleset.multiplier;
    }
  });

  return totalTOIL;
}

// New function to calculate hours from raw time values
function calculateHoursFromValues(startValue: number, endValue: number): number {
  // Convert the raw values to hours, then subtract to get the overlapping interval
  return endValue - startValue;
}
```

Reply:
This was like halfway but ty to code AI stuff, it now handles overlap if the person say has a rule for 9am-5pm and the hours they do are 8am-10am, it will now calculate 1 hour of time owed (before it was 0) so yea and same thing for if you go beyond the rule e.g 4-6pm would be 1 hour owing based off the rules